File: Java/src/test/java/com/cryptofundamentals/HashTest.java
package com.cryptofundamentals;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class HashTest {
    
    @Test
    public void testHashMessage() throws Exception {
        String message = "Alice knows Bob's secret";

        byte[] digest = Crypto.computeHash(message);

        assertEquals(64, digest.length);
    }

    @Test
    public void testPasswordBasedKeyDerivationFunction() throws Exception {
        String passphrase = "Twas brillig and the slithy toves did gire and gimble in the wabe";
        byte[] salt = Crypto.generateSalt();

        byte[] key = Crypto.deriveKey(passphrase, salt);

        assertEquals(32, key.length);
    }
}

File: .NET/CryptoHelpers/CryptoHelpers.Test/HashTest.cs
using System;
using Xunit;
using FluentAssertions;

namespace CryptoHelpers.Test
{
    public class HashTest
    {
        [Fact]
        public void HashAMessage()
        {
            string message = "Alice knows Bob's secret.";

            byte[] digest = HashHelpers.ComputeHash(message);

            digest.Length.Should().Be(64);
        }

        [Fact]
        public void PasswordBasedKeyDerivationFunction()
        {
            string passphrase = "Twas brillig and the slithy toves did gyre and gimble in the wabe";
            byte[] salt = HashHelpers.GenerateSalt();

            byte[] key = HashHelpers.DeriveKey(passphrase, salt);

            key.Length.Should().Be(32);
        }
    }
}
